added the method supportedFiltersAndFields()

jwbargsten opened the pull request:
For newcomers (like me) it is confusing which filter corresponds to
which field. I added a function to give an overview in form of a
data.frame.

Greetings from the CSAMA 2017. I hope it is useful.

jwbargsten committed:
added the method supportedFiltersAndFields()

For newcomers (like me) it is confusing which filter corresponds to
which field. I added a function to give an overview in form of a
data.frame.

jorainer commented:
Thanks @jwbargsten . I think that's a useful addition (after some modifications). @mtmorgan any comments?

mtmorgan commented:
Sounds like a good idea to me; maybe remove suppportedFilters(), or change the output so that the user isn't overwhelmed with too many similar choices?

jorainer commented:
OK, then I'll review the pull request.

jorainer commented on R/AnnotationFilter.R:
```diff
@@ -373,6 +376,20 @@ NULL
+.supportedFiltersAndFields <- function() {
+  filters <- supportedFilters()
+  df <- data.frame(filter=filters, field=.classToField(filters))
+  df
```
One thing: the mapping of `GRangesFilter` to a field `g_ranges` will not work - I suggest to list an `NA` for `GRangesFilter`.

jorainer reviewed (commented):
1) replace the output of `supportedFilters` with the output of your function.
2) Add an unit test.
3) Ensure that `R CMD check` completes without errors or warnings.

jwbargsten committed:
merged supportedFilers() and supportedFiltersAndFields()

now supportedFilters() will return a data frame with mappings.
Added also .FILTERS_WO_FIELDS variable to deal with special cases.

jorainer reviewed (approved):
Nice job @jwbargsten . Please send me your email, so I can add you to the contributors.

jwbargsten commented:
Thanks. My mail is jw@bargsten.org
